iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/extensions/RemoteReactivePowerControlImpl.java

```diff
@@ -7,23 +7,25 @@
  */
 package com.powsybl.iidm.network.impl.extensions;
 
+import com.powsybl.commons.PowsyblException;
 import com.powsybl.commons.util.trove.TBooleanArrayList;
 import com.powsybl.iidm.network.Generator;
 import com.powsybl.iidm.network.Terminal;
 import com.powsybl.iidm.network.extensions.RemoteReactivePowerControl;
 import com.powsybl.iidm.network.impl.AbstractMultiVariantIdentifiableExtension;
+import com.powsybl.iidm.network.impl.TerminalExt;
 import gnu.trove.list.array.TDoubleArrayList;
 
 /**
  * @author Bertrand Rix {@literal <bertrand.rix at artelys.com>}
  */
 public class RemoteReactivePowerControlImpl extends AbstractMultiVariantIdentifiableExtension<Generator> implements RemoteReactivePowerControl {
 
-    private TDoubleArrayList targetQ;
+    private final TDoubleArrayList targetQ;
 
-    private final Terminal regulatingTerminal;
+    private Terminal regulatingTerminal;
```

Comment:
we should probably add `setRegulatingTerminal(Terminal)` on `RemoteReactivePowerControl` interface. Not sure why it wasn't done before.
Now we could leave an extension completely ineffective with `regulatingTerminal` set to `null` on removal of referenced connectable.

Reply:
This is a good point. We cannot like for RegulatingPoint to automatically fallback to a local terminal, so maybe we should  remove the extension to avoid an inconsistent state of the extension ? and also indeed to way to change the terminal so let the user change the terminal before removing the referenced element.

Reply:
So, I chose to remove the extension with a warning.

Reply:
> So, I chose to remove the extension with a warning.

This is the way. thanks!